src/main/java/seedu/duke/commands/TransactionCommand.java

```diff
@@ -0,0 +1,161 @@
+        ui.printMessage(transactionsString);
+    }
+
+    /**
+     * Method that searches a string for a substring
+     * Upon finding that substring, returns the rest of the string after that substring,
+     * excluding the searched substring
+     *
+     * @param stringToSearch The string to search within
+     * @param substring      The substring to search for
+     * @return The rest of the input string after and excluding the input substring, null if substring is not found
+     */
+    private String findSubstringToEndAfterASubstring (String stringToSearch, String substring) {
+        int substringStartIndex = stringToSearch.indexOf(substring);
+        if (substringStartIndex == -1) {
+            return null;
+        }
+        return stringToSearch.substring(stringToSearch.indexOf(" ", substringStartIndex)).trim();
+    }
+
+    /**
+     * Method that prints transactions within some search parameters
+     * The search parameters are determined by the flag passed in
+     *
+     * @param ui   The instance of the UI class
+     * @param args The String array representing the user input split by space
+     * @param flag The String flag determining which search parameters are used
+     * @throws InvalidSearchTransactionByDescException     if the description given to search by is invalid
+     * @throws NoTransactionsRecordedException             if there are no transactions in the list
+     * @throws InvalidSearchTransactionByMonthException    if the month given to search by is invalid
+     * @throws InvalidSearchTransactionByDateException     if the date given to search by is invalid
+     * @throws InvalidSearchTransactionByCurrencyException if the currency given to search by is invalid
+     * @throws NoTransactionsOfSearchParameterException    if there are no transactions matching search parameters
+     * @throws InvalidTransactionFlagException             if the flag given is invalid
+     * @throws IllegalArgumentException                    if the currency given to search by is invalid
+     */
+    private void printTransactionsByFlag (Ui ui, String[] args, String flag) throws
+            InvalidSearchTransactionByDescException, NoTransactionsRecordedException,
+            InvalidSearchTransactionByMonthException, InvalidSearchTransactionByDateException,
+            InvalidSearchTransactionByCurrencyException, NoTransactionsOfSearchParameterException,
+            InvalidTransactionFlagException, IllegalArgumentException {
+        TransactionManager transactionManager = TransactionManager.getInstance();
+        String transactionsString;
+        switch (flag) {
+        case TransactionFlag.TRANSACTION_DESC_FLAG:
+            if (args.length < 3) {
+                throw new InvalidSearchTransactionByDescException();
+            }
+            String description = findSubstringToEndAfterASubstring(input, TransactionFlag.TRANSACTION_DESC_FLAG);
```

Comment:
i think this suggestion does the same thing too.  Just FYI, current method has no problem. 
```suggestion
            String description = input.split("/(TransactionFlag.TRANSACTION_DESC_FLAG)", 2 )[1].trim();
```

Reply:
Good suggestion, though I think the current implementation abstracts it a bit more cleanly